src/mpid/ch4/include/mpidpre.h

```diff
@@ -159,6 +159,8 @@ typedef struct MPIDI_CH4U_req_ext_t {
 typedef struct MPIDI_CH4U_req_t {
     union {
     MPIDI_NM_REQUEST_AM_DECL} netmod_am;
+    union {
+    MPIDI_SHM_REQUEST_AM_DECL} shm_am;
```

Comment:
Shouldn't the netmod_am and shm_am be inside a single union?  We'll only use one at a time (whichever is the transport).